config/taxonomy/application.values.schema.json

```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "application.values.schema.json",
+    "title": "Application Values Taxonomy",
+    "description": "Values contributed to by the mesh for data applications.",
+    "definitions": {
+        "intent": {"$ref": "policymanager.structs.schema.json#/definitions/intent"},
+        "role": {"$ref": "policymanager.structs.schema.json#/definitions/role"},
+        "app_info": {
+            "type": "object",
+            "properties": {
+                "intent": { "$ref": "policymanager.structs.schema.json#/definitions/intent"},
+                "role": { "$ref": "policymanager.structs.schema.json#/definitions/role"}
+            },
+            "required": ["intent"]
+        },
+        "interface": {
```

Comment:
You should factor out the interface to another file as it should be check for m4dmodule as well.
See Source and Sink definitions in [m4dmodules_types.go](https://github.com/IBM/the-mesh-for-data/blob/master/manager/apis/app/v1alpha1/m4dmodule_types.go#L50)

Reply:
You can refer to the `interface` in this file by pointing to `application.values.schema.json#/definitions/interface` from another schema file.

Reply:
Right, and I actually gave the wrong motivation.
Down the road interface would be generated instead of manually written.
When a new m4dmodule is applied we would re-generate the values scheme for interface adding new protocols that were newly added in the m4module. For example, someone add a m4module with protocol REST and data_format JSON, we would add these values to the validation scheme for interface.
So it would be good that the ```$ref``` be the other way.

Reply:
I added the `interface` schema to `module.values.schema.json` and am referring to it from `application.values.schema.json` now